File: frontends/cri-web-micr/lib/session.ts
"use client";

import { useEffect, useState } from "react";
import { useRouter } from "next/navigation";

// Sesión-MVP: la cédula del ciudadano activo se guarda en localStorage.
// En producción esto vendría de un JWT validado por el gateway.

export function useActiveCedula(): { cedula: string | null; loading: boolean } {
  const router = useRouter();
  const [cedula, setCedula] = useState<string | null>(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const c = window.localStorage.getItem("micr_cedula");
    if (!c) {
      router.replace("/");
      return;
    }
    setCedula(c);
    setLoading(false);
  }, [router]);

  return { cedula, loading };
}

export function clearSession() {
  if (typeof window !== "undefined") {
    window.localStorage.removeItem("micr_cedula");
  }
}

File: frontends/cri-mobile/lib/session.dart
// Sesión-MVP: la cédula del ciudadano activo se guarda en shared_preferences.
// En producción esto vendría de un JWT validado por el gateway.

import 'package:shared_preferences/shared_preferences.dart';

class Session {
  static const _key = 'micr_cedula';

  static Future<String?> activeCedula() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString(_key);
  }

  static Future<void> setActiveCedula(String cedula) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(_key, cedula);
  }

  static Future<void> clear() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove(_key);
  }
}
